torch/nn/parallel/distributed.py

```diff
@@ -673,6 +673,7 @@ def no_sync(self):
             self.require_backward_grad_sync = old_require_backward_grad_sync
 
     def forward(self, *inputs, **kwargs):
+        self.reducer.save_thread_local_state()
```

Comment:
multi threads could be spawned after this, will it have problems? e.g., single process multiple services will run multi threads to compute outputs afterwards. 

just curious, what kind of thread local states are saved here (e.g., profiler enable flag and else?)? is it large?

Reply:
I don't think there's a problem if multiple threads are spawned in forward after this, if those are spawned with `torch.jit.fork` which is the recommendation for inter-op parallelism, then the profiler enabled flag should be carried over transparently.

the full structure is in `aten/ThreadLocalState.h`, the flags are not very large, it's basically dispatch key, thread-local debug info struct (pointer to debug info struct), and an instance of `RecordFunctiontls` specifying the callbacks to be run.

Reply:
sounds good, thanks for clarification